crates/primitives/src/transaction/tx_type.rs

```diff
@@ -181,6 +182,30 @@ impl PartialEq<TxType> for u8 {
     }
 }
 
+impl Encodable for TxType {
+    fn encode(&self, out: &mut dyn bytes::BufMut) {
+        (*self as u8).encode(out);
+    }
+
+    fn length(&self) -> usize {
+        1
+    }
+}
+
+impl Decodable for TxType {
+    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
+        if buf.is_empty() {
+            return Err(alloy_rlp::Error::InputTooShort)
+        } else if buf.len() > 1 {
+            return Err(alloy_rlp::Error::Overflow)
+        }
+
+        let ty = u8::decode(buf)?;
```

Comment:
surprised that these checks are not already done in the `u8` decode impl

Reply:
they are! thanks. checking out a lot of stuff on the fly from emhane/op-import-receipts to shrink its scope